extensions/ezmsg-sigproc/ezmsg/sigproc/synth.py

```diff
@@ -37,9 +37,13 @@ async def on_settings(self, msg: ClockSettings) -> None:
 
     @ez.publisher(OUTPUT_CLOCK)
     async def generate(self) -> AsyncGenerator:
+        t_0 = time.time()
+        n_dispatch = 0
         while True:
             if self.STATE.cur_settings.dispatch_rate is not None:
-                await asyncio.sleep(1.0 / self.STATE.cur_settings.dispatch_rate)
+                n_dispatch += 1
```

Comment:
Although this will suffer from floating point accuracy issues if the code runs long enough, this is definitely more accurate than what I had before.  If folks complain down the road, we could add an occasional reset of t_0 and n_dispatch to keep them below some threshold to avoid floating point inaccuracies.